portal-bridge/src/types/state/content.rs

```diff
@@ -42,3 +45,130 @@ pub fn create_content_value(
         },
     ))
 }
+
+pub fn create_contract_content_key(
+    address: Address,
+    code_hash: B256,
+) -> anyhow::Result<StateContentKey> {
+    Ok(StateContentKey::ContractBytecode(ContractBytecodeKey {
+        address,
+        code_hash,
+    }))
+}
+
+pub fn create_contract_content_value(
+    block_hash: B256,
+    account_proof: &AccountProof,
+    code: Bytecode,
+) -> anyhow::Result<StateContentValue> {
+    let account_proof = TrieProof::from(
+        account_proof
+            .proof
+            .iter()
+            .map(|bytes| EncodedTrieNode::from(bytes.to_vec()))
+            .collect::<Vec<EncodedTrieNode>>(),
+    );
+    let code = code.bytecode.to_vec().into();
+    Ok(StateContentValue::ContractBytecodeWithProof(
+        ContractBytecodeWithProof {
+            block_hash,
+            code,
+            account_proof,
+        },
+    ))
+}
+
+pub fn create_storage_content_key(
+    storage_proof: &AccountProof,
+    address: Address,
+) -> anyhow::Result<StateContentKey> {
+    let last_node = storage_proof
+        .proof
+        .last()
+        .ok_or(anyhow!("Missing proof!"))?;
+    let last_node_hash = keccak256(last_node);
+
+    Ok(StateContentKey::ContractStorageTrieNode(
+        ContractStorageTrieNodeKey {
+            path: Nibbles::try_from_unpacked_nibbles(&storage_proof.path)?,
+            node_hash: last_node_hash,
+            address,
+        },
+    ))
+}
+
+pub fn create_storage_content_value(
+    block_hash: B256,
+    account_proof: &AccountProof,
+    storage_proof: &AccountProof,
+) -> anyhow::Result<StateContentValue> {
+    let account_proof = TrieProof::from(
+        account_proof
+            .proof
+            .iter()
+            .map(|bytes| EncodedTrieNode::from(bytes.to_vec()))
+            .collect::<Vec<EncodedTrieNode>>(),
+    );
+
+    let storage_proof = TrieProof::from(
+        storage_proof
+            .proof
+            .iter()
+            .map(|bytes| EncodedTrieNode::from(bytes.to_vec()))
+            .collect::<Vec<EncodedTrieNode>>(),
+    );
+    Ok(StateContentValue::ContractStorageTrieNodeWithProof(
+        ContractStorageTrieNodeWithProof {
+            block_hash,
+            storage_proof,
+            account_proof,
+        },
+    ))
+}
+
+pub fn full_key_path_to_address_hash(key_path: &[u8]) -> B256 {
```

Comment:
That doesn't make sense to me as we are getting the address's hash or keccak(addressO

Reply:
Would this ever be used for something else though? I think we should give it a generic name if we ever use it in a generic way

Reply:
I think it's generic enough that I would even extracted into separate file (e.g. `utils.rs`). Reason is that if I'm writing new code and I feel like there might be generic function that I need to use (and I would say this would fall into that category), I'm going to look for it in some library or some `utils.rs` or similar, or by searching by name (and I'm not likely going to find it inside `content.rs` with `full_key_path_to_address_hash` name). Of course this assumes that when you are writing code, you are not familiar with the entire codebase, which is the case in general.

So it is good coding practice to extract generic functions and give them generic names, even if they are currently used only in one place.

I'm not too keen on this, so it's up to you.

Also, you didn't address other comments regarding checking the validity of the input (length can be 63, bytes can have invalid values), which I would might be useful/good even if it is not generic.

Reply:
> Also, you didn't address other comments regarding checking the validity of the input (length can be 63, bytes can have invalid values), which I would might be useful/good even if it is not generic.

I will add this as I think it is good

Reply:
I will move it to utils.rs. I will change the name from `full_key_path_to_address_hash` to ` full_nibble_path_to_address_hash`. I see this function as an inversion to the util function `address_to_nibble_path`. I want to keep `address_hash` in the name because I want when people read `full_nibble_path_to_address_hash` to understand what it is used for. I find `full_nibble_path_to_packed_path` very confusing because the name is so generic I don't understand what it would be used for when reading it.

So if we find a more generic usecase for it in the future we can update it to the generic name. But for now I will do the inbetween name which still contains `address_hash` as you said it is up to me. We still maintain the possibility of changing it depending on what the future brings